src/components/channellist/channellisttree.cpp

```diff
@@ -636,6 +636,12 @@ void ChannelListTree::OnVoiceUserDisconnect(Snowflake user_id, Snowflake channel
     if (auto iter = GetIteratorForRowFromIDOfType(user_id, RenderType::VoiceParticipant)) {
         m_model->erase(iter);
     }
+
+    auto& abaddon = Abaddon::Get();
+    auto ssrc = abaddon.GetDiscordClient().GetSSRCOfUser(user_id);
+    if (ssrc) {
+        abaddon.GetAudio().RemoveSSRC(*ssrc);
+    }
```

Comment:
not the right place to do this. the signal is emitted here:
https://github.com/uowuo/abaddon/blob/ada3420dce68a3f4f5a3f1bcaf55f8d688b3d27f/src/discord/discord.cpp#L2389-L2404
if we wanna avoid duplication maybe we can split out another function in DiscordClient for disconnects but i dont think (effectively) 2 lines of duplication is a big deal

Reply:
I suppose it would be better to instead make `VoiceAudio` connect to these signals and handle them there? For now I just redirect methods from `AudioManager` to stay compatible.

Reply:
connecting there is probably fine. cant think of a better place off top of my head just not in the ui code lul